src/uwtools/drivers/driver.py

```diff
@@ -77,8 +98,85 @@ def schema_file(self) -> str:
         The path to the file containing the schema to validate the config file against.
         """
 
+    @staticmethod
+    def stage_files(
+        run_directory: str, files_to_stage: Dict[str, Union[list, str]], link_files: bool = False
+    ) -> None:
+        """
+        Creates destination files in run directory and copies or links contents from the source path
+        provided.
+
+        :param run_directory: Path of desired run directory.
+        :param files_to_stage: File names in the run directory (keys) and their source paths
+            (values).
+        :param link_files: Whether to link or copy the files.
+        """
+        link_or_copy = os.symlink if link_files else shutil.copyfile
+        for dst_fn, src_path in files_to_stage.items():
+            dst_path = os.path.join(run_directory, dst_fn)
+            if isinstance(src_path, list):
+                Driver.stage_files(
+                    dst_path,
+                    {os.path.basename(src): src for src in src_path},
+                    link_files,
+                )
+                continue
```

Comment:
I added support for a list of files that needed to be linked in with their original names, but those names contained configurable parts. Not sure if that's the right thing to do.

Reply:
re: but those names contained configurable parts

For cycle-dependent files, we might need a file named something like `rrfs.t00z.gfs_bndy.tile7.f000.nc`...we might need every cycle to have a different name linked in. So `t00z` and `f000` wouldn't be known until run-time, so we couldn't know the key to put in the dict beforehand.

Reply:
Seems fine, but perhaps orthogonal? Might one way to deal with this be to follow a rule that says:

> Whe linking/copying cycle-dependent files, assume that destination names taken from YAML might contain tokens like `{hh}` (cycle hour) or `{ttt}` (leadtime), so always call `.format()` on those names before using them, which will be a no-op if the files do _not_ contain time-related tokens.

Do we already have support for time-templated filenames, or were you just thinking ahead and mentioning a future need. Sorry if I missed it...

Reply:
We do not yet have that functionality. I put it into a branch with other work I've been doing, but I wasn't convinced it needed to be in scope for this PR.